### PR TITLE
Search persistence UI

### PR DESCRIPTION
**Pull request by microbit-robert**
Add UI to show existing search query in header. 

See #440

### CODE REVIEW

File: src/workbench/SideBarHeader.tsx [hunk 1]
```diff
@@ -42,15 +44,26 @@ const SideBarHeader = () => {
       },
       [search]
     );
+  const handleClear = useCallback(() => {
+    setQuery("");
+    setResults(undefined);
+    if (ref.current) {
+      ref.current.focus();
```
Comment (microbit-matt-hillsdon) [inline src/workbench/SideBarHeader.tsx:51]: This isn't focusing the same thing it was before.

File: src/documentation/Search.tsx [hunk 2]
```diff
@@ -30,15 +31,9 @@ const Search = ({
...
+  onClear,
 }: SearchProps) => {
   const ref = useRef<HTMLInputElement>(null);
-  const handleClear = useCallback(() => {
-    setQuery("");
-    setResults(undefined);
-    if (ref.current) {
-      ref.current.focus();
-    }
-  }, [setQuery, setResults]);
```
Comment (microbit-matt-hillsdon) [inline src/documentation/Search.tsx:41]: Let's try keeping this to do the focus handling then call onClear for the parent state to update.

File: src/workbench/SideBarHeader.tsx [hunk 3]
```diff
@@ -133,6 +147,42 @@ const SideBarHeader = () => {
             Search
           </Button>
         )}
+        {flags.search && query && (
+          <ButtonGroup
+            isAttached
+            backgroundColor="white"
+            width="full"
+            borderRadius="3xl"
```
Comment (microbit-matt-hillsdon) [inline src/workbench/SideBarHeader.tsx:155]: The focus outline looks very odd. Let's try a pattern similar to https://chakra-ui.com/docs/form/input so the focus ring is around the whole element and the other button is visually inside (via absolute positioning).

**Review by microbit-matt-hillsdon — CHANGES_REQUESTED** [review]
Minor tweaks but looks good.

File: src/workbench/SideBarHeader.tsx [hunk 4]
```diff
@@ -48,15 +50,29 @@ const SideBarHeader = () => {
...
-    ? ref.current!.clientWidth - offset - 14 + "px"
+  const width = sidebarWidth.current
+    ? sidebarWidth.current!.clientWidth - offset - 14 + "px"
     : undefined;
+
+  const { sizedPaneSize } = useSplitViewContext();
+  const [searchWidth, setSearchWidth] = useState<number | undefined>();
+  useEffect(() => {
+    setSearchWidth((sizedPaneSize || 0) - offset - 14);
+  }, [offset, sizedPaneSize]);
```
Comment (microbit-robert) [inline src/workbench/SideBarHeader.tsx:70]: We need a better way than this. It causes so much lag on moving the divider.

The problem I was trying to solve was having a long search query when the search dialog is closed. Avoid text or it's container spilling over into the editor. Avoid this happening when the user moves the split view divider. I think we need a `maxWidth` from somewhere. Putting `overflow="hidden"` in the wrong place also hides the focus ring.

Comment (microbit-robert) [reply src/workbench/SideBarHeader.tsx]: I think I just pushed a fix for it, which was simple, and I was overcomplicating it.

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Pull request merged**